src/core/variable.jl

```diff
@@ -571,51 +602,63 @@ function variabe_mc_on_off_storage_active(pm::_PMs.AbstractPowerModel; nw::Int=p
         inj_lb[cnd], inj_ub[cnd] = _PMs.ref_calc_storage_injection_bounds(_PMs.ref(pm, nw, :storage), _PMs.ref(pm, nw, :bus), cnd)
     end
 
-    _PMs.var(pm, nw)[:ps] = Dict(i => JuMP.@variable(pm.model,
+    ps = _PMs.var(pm, nw)[:ps] = Dict(i => JuMP.@variable(pm.model,
         [cnd in 1:ncnds], base_name="$(nw)_ps_$(i)",
         lower_bound = inj_lb[cnd][i],
         upper_bound = inj_ub[cnd][i],
         start = comp_start_value(_PMs.ref(pm, nw, :storage, i), "ps_start", cnd)
     ) for i in _PMs.ids(pm, nw, :storage))
+
+    report && _PMs.sol_component_value(pm, nw, :storage, :ps, _PMs.ids(pm, nw, :storage), ps)
 end
 
 
 "Create variables for `reactive` storage injection"
-function variable_mc_on_off_storage_reactive(pm::_PMs.AbstractPowerModel; nw::Int=pm.cnw)
+function variable_mc_on_off_storage_reactive(pm::_PMs.AbstractPowerModel; nw::Int=pm.cnw, report::Bool=true)
+    cnds = _PMs.conductor_ids(pm; nw=nw)
     ncnds = length(_PMs.conductor_ids(pm, nw))
 
-    _PMs.var(pm, nw)[:qs] = Dict(i => JuMP.@variable(pm.model,
+    qs = _PMs.var(pm, nw)[:qs] = Dict(i => JuMP.@variable(pm.model,
         [cnd in 1:ncnds], base_name="$(nw)_qs_$(i)",
         lower_bound = min(0, _PMs.ref(pm, nw, :storage, i, "qmin", cnd)),
         upper_bound = max(0, _PMs.ref(pm, nw, :storage, i, "qmax", cnd)),
         start = comp_start_value(_PMs.ref(pm, nw, :storage, i), "qs_start", cnd)
     ) for i in _PMs.ids(pm, nw, :storage))
+
+    report && _PMs.sol_component_value(pm, nw, :storage, :qs, _PMs.ids(pm, nw, :storage), qs)
 end
 
 
 "voltage variable magnitude squared (relaxed form)"
-function variable_mc_voltage_magnitude_sqr_on_off(pm::_PMs.AbstractPowerModel; nw::Int=pm.cnw)
+function variable_mc_voltage_magnitude_sqr_on_off(pm::_PMs.AbstractPowerModel; nw::Int=pm.cnw, report::Bool=true)
+    cnds = _PMs.conductor_ids(pm; nw=nw)
     ncnds = length(_PMs.conductor_ids(pm, nw))
 
-    _PMs.var(pm, nw)[:w] = Dict(i => JuMP.@variable(pm.model,
+    w = _PMs.var(pm, nw)[:w] = Dict(i => JuMP.@variable(pm.model,
         [cnd in 1:ncnds], base_name="$(nw)_w_$(i)",
         lower_bound = 0,
         upper_bound = _PMs.ref(pm, nw, :bus, i, "vmax", cnd)^2,
```

Comment:
`_PMs.ref(pm, nw, :bus, i, "vmax", cnd)` => `_PMs.ref(pm, nw, :bus, i, "vmax")[cnd]`

This is my fault actually, I created a new ref method to support this. When all other unit tests work, I will remove this overloading of ref and I will clean up the code.

Reply:
ok